pkg/distributor/query.go

```diff
@@ -277,20 +297,24 @@ func (d *Distributor) queryIngesterStream(ctx context.Context, replicationSets [
 			} else if len(resp.Chunkseries) > 0 {
 				// Enforce the max chunks limits.
 				if err := queryLimiter.AddChunks(ingester_client.ChunksCount(resp.Chunkseries)); err != nil {
+					result.freeBuffers()
```

Comment:
We could use a single defer statement that checks the `err` and frees if it's not nil, wdyt? Then we don't have to care about making sure that all cases are covered.

Reply:
Yeah, I thought one _could_ also use `defer`. If you prefer it, I can do it.

Reply:
Done, PTAL.